tests/unit/test_space.py

```diff
@@ -11,18 +11,47 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+from __future__ import annotations
+
 import copy
 import itertools
 from typing import Container, FrozenSet, Tuple
 
 import numpy.testing as npt
 import pytest
 import tensorflow as tf
+from typing_extensions import Final
 
 from tests.util.misc import TF_DEBUGGING_ERROR_TYPES, ShapeLike, various_shapes
 from trieste.space import Box, DiscreteSearchSpace, SearchSpace
 
 
+class Integers(SearchSpace):
+    def __init__(self, exclusive_limit: int):
+        assert exclusive_limit > 0
+        self.limit: Final[int] = exclusive_limit
+
+    def sample(self, num_samples: int) -> tf.Tensor:
+        return tf.random.shuffle(tf.range(self.limit))[0]
+
+    def __contains__(self, point: tf.Tensor) -> tf.Tensor:
```

Comment:
what happens if you test it with point=0.1?

Reply:
good catch. probably not the right thing

Reply:
ok, fixed I think